src/types.ts

```diff
@@ -18,3 +18,9 @@ export type SerializedInput<Node = unknown, LinkProps = unknown> = {
 };
 
 export type NoInfer<T> = [T][T extends any ? 0 : never];
+
+export type WeightParams = {
```

Comment:
Maybe be more specific with NextWeightFnParams ?

Reply:
This is done